src/Codec/Archive/Zip.hs

```diff
@@ -399,25 +419,35 @@ writeEntry opts entry = do
 
 #ifndef _WINDOWS
 -- | Write an 'Entry' representing a symbolic link to a file.
--- If the 'Entry' does not represent a symbolic link or
--- the options do not contain 'OptPreserveSymbolicLinks`, this
--- function behaves like `writeEntry`.
+-- If the options do not contain 'OptPreserveSymbolicLinks`, this
+-- function is a noop.
 writeSymbolicLinkEntry :: [ZipOption] -> Entry -> IO ()
 writeSymbolicLinkEntry opts entry =
-  if OptPreserveSymbolicLinks `notElem` opts
-     then writeEntry opts entry
-     else do
-        if isEntrySymbolicLink entry
-           then do
-             let prefixPath = case [d | OptDestination d <- opts] of
-                                   (x:_) -> x
-                                   _     -> ""
-             let targetPath = fromJust . symbolicLinkEntryTarget $ entry
-             let symlinkPath = prefixPath </> eRelativePath entry
-             when (OptVerbose `elem` opts) $ do
-               hPutStrLn stderr $ "linking " ++ symlinkPath ++ " to " ++ targetPath
-             createSymbolicLink targetPath symlinkPath
-           else writeEntry opts entry
+  case [s | OptSymbolicLinks s <- opts] of
+     (symOpt:_) -> do
+        let prefixPath = case [d | OptDestination d <- opts] of
+                              (x:_) -> x
+                              _     -> ""
+        let targetPath = fromJust . symbolicLinkEntryTarget $ entry
+        let symlinkPath = prefixPath </> eRelativePath entry
+        when (OptVerbose `elem` opts) $ do
+          let logLine = if symOpt == OptSymlinkClobber
+                then "forcefully linking " ++ symlinkPath ++ " to " ++ targetPath
+                else "linking " ++ symlinkPath ++ " to " ++ targetPath
+          hPutStrLn stderr logLine
```

Comment:
Instead of having this distinction in log messages between "linking" and "forcefully linking", I think it would be better to keep the original "linking," but issue an additional message (in forceSymLink) if an existing link is removed ("removing existing symlink ...").